src/include/homestore/btree/detail/simple_node.hpp

```diff
@@ -147,8 +147,7 @@ class SimpleNode : public BtreeNode {
     }
 
     uint32_t num_entries_by_size(uint32_t start_idx, uint32_t size) const override {
-        uint32_t possible_entries = (size == 0) ? 0 : (size - 1) / get_nth_obj_size(0) + 1;
-        return std::min(possible_entries, this->total_entries() - start_idx);
+        return std::min(size / get_nth_obj_size(0), this->total_entries() - start_idx);
```

Comment:
for simple node type, when the size was not divisible, the round up value is returned which can lead to out of space happens in merge. 

For example:
```
size = 100
get_nth_obj_size = 30
```

The correct return value must be `100/30 = 3` and not 4.